### PR TITLE
Repaired hla_staging branch manually

### PR DESCRIPTION
**Pull request by stsci-hack**
This set of changes should include all the changes that were made to the hla_staging branch without contamination from master.  All subsequent PRs for the hla-staging branch should start from this PR after it has been merged.

All changes from the following closed PRs were verified and included in this PR: #304, #307, #308, #315, #317, #318, #321, #323, #327, #328, #330, #332, #333, #334.

### CODE REVIEW

File: drizzlepac/alignimages.py [hunk 1]
```diff
@@ -267,17 +262,14 @@ def run_align(input_list, archive=False, clobber=False, debug=False, update_hdr_
...
+    log.info("*** HLAPIPELINE Processing Version {!s} ({!s}) started at: {!s} ***\n".format(__version__, __version_date__, util._ptime()[0]))
 
     # Define astrometric catalog list in priority order
     catalog_list = ['GAIADR2', 'GAIADR1']
 
     # 0: print git info
     if print_git_info:
-        log.info(
-            "{} STEP 0: Display Git revision info {}".format("-" * 20, "-" * 48))
+        log.info("-------------------- STEP 0: Display Git revision info  ------------------------------------------------")
```
Comment (mdlpstsci) [inline drizzlepac/alignimages.py]: Looks like this is the old way to display the divider line which violates PEP8 line length.  Keep Mike's new and improved way.

File: drizzlepac/alignimages.py [hunk 2]
```diff
@@ -1174,42 +1141,37 @@ def update_image_wcs_info(tweakwcs_output):
...
+                                 reusename=True, verbose=True)
         if chipctr == num_sci_ext:
             # Close updated flc.fits or flt.fits file
+            # log.info("CLOSE {}\n".format(image_name))  # TODO: Remove before deployment
             hdulist.flush()
             hdulist.close()
 
             # Create headerlet
-            out_headerlet = headerlet.create_headerlet(img_name, hdrname=wcs_name, wcsname=wcs_name)
+            out_headerlet = headerlet.create_headerlet(image_name, hdrname=wcs_name, wcs_name=wcs_name)
```
Comment (mdlpstsci) [inline drizzlepac/alignimages.py]: Please check the parameter is `wcs_name` and not `wcsname`.  I looked at _what I believe_ to be the version of stwcs we are using and the parameter is `wcsname`.

**Comment by stsci-hack** [comment]
This code has been used to successfully run the test_apriori tests after 3f91c80 was included and is ready for further review.

**Comment by dulude** [comment]
> @dulude : This code has been used to successfully run the test_apriori tests after [3f91c80](https://github.com/spacetelescope/drizzlepac/commit/3f91c8011b60ec9c84a0beddcee05658f533c4fa) was included and is ready for further review.

Thanks. I'll begin my review.

File: drizzlepac/alignimages.py [hunk 3]
```diff
@@ -622,27 +594,26 @@ def run_align(input_list, archive=False, clobber=False, debug=False, update_hdr_
...
                     filtered_table[index]['rotation'] = item.meta['fit_info']['rot']
 
-                    # populate filtered_table fields "status", "compromised" and "processMsg" with
-                    # fit_status_dict fields "valid", "compromised" and "reason".
+                    # populate filtered_table fields "status", "compromised" and
+                    # "processMsg" with fit_status_dict fields "valid", "compromised"
+                    # and "reason".
                     explicit_dict_key = "{},{}".format(item.meta['name'], item.meta['chip'])
-                    if fit_status_dict[explicit_dict_key]['valid'] is True:
+                    if fit_status_dict[explicit_dict_key]['valid'] == True:
```
Comment (dulude) [inline drizzlepac/alignimages.py]: '==' should be 'is'.

File: drizzlepac/alignimages.py [hunk 4]
```diff
@@ -622,27 +594,26 @@ def run_align(input_list, archive=False, clobber=False, debug=False, update_hdr_
...
                         filtered_table[index]['status'] = 0
                     else:
                         filtered_table[index]['status'] = 1
-
-                    if fit_status_dict[explicit_dict_key]['compromised'] is False:
+                    if fit_status_dict[explicit_dict_key]['compromised'] == False:
```
Comment (dulude) [inline drizzlepac/alignimages.py]: '==' should be 'is'.

File: drizzlepac/alignimages.py [hunk 5]
```diff
@@ -864,7 +837,7 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
     overall_valid = True
     overall_comp = False
     for item in imglist:
-        if not item.meta['fit_info']['status'].startswith('FAILED'):
+        if item.meta['fit_info']['status'].startswith('FAILED') != True:
```
Comment (dulude) [inline drizzlepac/alignimages.py]: "!=" operator should not be used with Booleans. Maybe replace "!= True" with "is False"

File: drizzlepac/alignimages.py [hunk 6]
```diff
@@ -918,46 +890,46 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
...
-            fit_status_dict[dict_key]['reason'] = "RMS too large (>150 mas)!"
-        elif radial_offset_check is False:  # Failed radial offset check
-            fit_status_dict[dict_key]['valid'] = False
-            fit_status_dict[dict_key]['compromised'] = False
-            fit_status_dict[dict_key]['reason'] = "Radial offset value too large!"
-        elif nmatches_check is False:  # Too few matches
-            fit_status_dict[dict_key]['valid'] = True
-            fit_status_dict[dict_key]['compromised'] = True
-            fit_status_dict[dict_key]['reason'] = "Too few matches!"
+        if consistencyCheck == False:  # Failed consistency check
```
Comment (dulude) [inline drizzlepac/alignimages.py]: '==' should be 'is', or rewrite whole line as` if not consistencyCheck:`

File: drizzlepac/alignimages.py [hunk 7]
```diff
@@ -918,46 +890,46 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
...
+            fit_status_dict[dictKey]['valid'] = False
+            fit_status_dict[dictKey]['compromised'] = False
+            fit_status_dict[dictKey]['reason'] = "Consistency violation!"
+        elif large_rms_check == False:  # RMS value(s) too large
```
Comment (dulude) [inline drizzlepac/alignimages.py]: '==' should be 'is'

File: drizzlepac/alignimages.py [hunk 8]
```diff
@@ -918,46 +890,46 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
...
+            fit_status_dict[dictKey]['valid'] = False
+            fit_status_dict[dictKey]['compromised'] = False
+            fit_status_dict[dictKey]['reason'] = "RMS too large (>150 mas)!"
+        elif radialOffsetCheck == False:  # Failed radial offset check
```
Comment (dulude) [inline drizzlepac/alignimages.py]: '==' should be 'is'

File: drizzlepac/alignimages.py [hunk 9]
```diff
@@ -918,46 +890,46 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
...
+            fit_status_dict[dictKey]['valid'] = False
+            fit_status_dict[dictKey]['compromised'] = False
+            fit_status_dict[dictKey]['reason'] = "Radial offset value too large!"
+        elif nmatchesCheck == False:  # Too few matches
```
Comment (dulude) [inline drizzlepac/alignimages.py]: '==' should be 'is'

File: drizzlepac/alignimages.py [hunk 10]
```diff
@@ -1000,29 +972,25 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
...
-        log.info(
-            "Total fit RMS value = {} mas greater than the "
-            "maximum threshold value {}.".format(max_rms_val, MAX_FIT_RMS))
+        log.info("Total fit RMS value = {} mas greater than the maximum threshold value {}.".format(max_rms_val, MAX_FIT_RMS))
     if not overall_valid:
-        log.info(
-            "The fit solution for some or all of the images is not valid.")
-    if max_rms_val > MAX_FIT_RMS or overall_valid is False:
+        log.info("The fit solution for some or all of the images is not valid.")
+    if max_rms_val > MAX_FIT_RMS or overall_valid == False:
```
Comment (dulude) [inline drizzlepac/alignimages.py]: `overall_valid == False` should be rewritten as `overall_valid is False`

File: drizzlepac/alignimages.py (hunk 3 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Good catch. It will be corrected...

File: drizzlepac/alignimages.py (hunk 4 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: It will be corrected...

File: drizzlepac/alignimages.py (hunk 5 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Will replace with 'is False'.

File: drizzlepac/alignimages.py [hunk 11]
```diff
@@ -873,13 +846,12 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
...
 
         # Build fit_status_dict entry
-        dict_key = "{},{}".format(image_name, chip_num)
-        # Initialize dictionary entry for current image/chip
-        fit_status_dict[dict_key] = {'valid': False,
-                                     'max_rms': max_rms_val,
-                                     'num_matches': num_xmatches,
-                                     'compromised': False,
-                                     'reason': ""}
+        dictKey = "{},{}".format(image_name, chip_num)
```
Comment (dulude) [inline drizzlepac/alignimages.py]: Variable should not be camelcase. Rename 'dictKey' to dict_key and find/replace the remaining instances.

File: drizzlepac/alignimages.py [hunk 12]
```diff
@@ -888,27 +860,27 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
...
             if catalogs_remaining:
                 log.warning(
-                    "Not enough cross matches found between astrometric "
+                    "Not enough cross matches found between astrometric"
                     "catalog and sources found in {}".format(image_name))
                 continue
 
         # Execute checks
-        nmatches_check = False
+        nmatchesCheck = False
```
Comment (dulude) [inline drizzlepac/alignimages.py]: Variable should not be camelcase. Rename 'nmatchesCheck' to 'nmatches_check and find/replace the remaining instances.

File: drizzlepac/alignimages.py (hunk 6 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: This syntax, and the same type of syntax used in this entire if...elif...else block will be changed to use "not <variable>".

File: drizzlepac/alignimages.py [hunk 13]
```diff
@@ -888,27 +860,27 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
...
-            nmatches_check = True
+            nmatchesCheck = True
 
-        radial_offset_check = False
-        radial_offset = math.sqrt(
-            float(item.meta['fit_info']['shift'][0]) ** 2 +
-            float(item.meta['fit_info']['shift'][0]) ** 2) * item.wcs.pscale  # radial offset in arssec
-        if float(num_xmatches) * 0.36 > 0.8 + (radial_offset / 10.0) ** 8:
-            radial_offset_check = True
+        radialOffsetCheck = False
```
Comment (dulude) [inline drizzlepac/alignimages.py]: Variable should not be camelcase. Rename 'radialOffsetCheck' to 'radial_offset_check' and find/replace the remaining instances.

File: drizzlepac/alignimages.py [hunk 14]
```diff
@@ -888,27 +860,27 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
...
+        radialOffset = math.sqrt(
```
Comment (dulude) [inline drizzlepac/alignimages.py]: Variable should not be camelcase. Rename 'radialOffset' to 'radial_offset' and find/replace the remaining instances.

File: drizzlepac/alignimages.py [hunk 15]
```diff
@@ -918,46 +890,46 @@ def determine_fit_quality(imglist, filtered_table, catalogs_remaining, print_fit
         # if fit_rms_val < max_rms_val:
         #     fitRmsCheck = True
 
-        consistency_check = True
+        consistencyCheck = True
```
Comment (dulude) [inline drizzlepac/alignimages.py]: Variable should not be camelcase. Rename 'consistencyCheck' to 'consistency_check' and find/replace the remaining instances.

File: drizzlepac/hlautils/analyze.py [hunk 16]
```diff
@@ -84,7 +83,7 @@ def analyze_data(inputFileList, **kwargs):
 
     fit_method = None  # Fit algorithm used for alignment
     catalog = None     # Astrometric catalog used for alignment
-    catalogSources = 0 # Number of astrometric catalog sources determined based upon coordinate overlap with image WCS
+    catalogSources = 0  # Number of astrometric catalog sources determined based upon coordinate overlap with image WCS
```
Comment (dulude) [inline drizzlepac/hlautils/analyze.py]: Variable should not be camelcase. Rename 'catalogSources' to 'catalog_sources' and find/replace the remaining instances.

File: drizzlepac/hlautils/analyze.py [hunk 17]
```diff
@@ -84,7 +83,7 @@ def analyze_data(inputFileList, **kwargs):
...
     foundSources = 0   # Number of sources detected in images
```
Comment (dulude) [inline drizzlepac/hlautils/analyze.py]: Variable should not be camelcase. Rename 'foundSources' to 'found_sources' and find/replace the remaining instances.

File: drizzlepac/hlautils/analyze.py [hunk 18]
```diff
@@ -84,7 +83,7 @@ def analyze_data(inputFileList, **kwargs):
```
Comment (dulude) [inline drizzlepac/hlautils/analyze.py]: Variable should not be camelcase. Rename 'acsFiltNameList' to 'acs_filt_name_list' and find/replace the remaining instances.

File: drizzlepac/hlautils/analyze.py [hunk 19]
```diff
@@ -84,7 +83,7 @@ def analyze_data(inputFileList, **kwargs):
 
     fit_method = None  # Fit algorithm used for alignment
     catalog = None     # Astrometric catalog used for alignment
-    catalogSources = 0 # Number of astrometric catalog sources determined based upon coordinate overlap with image WCS
+    catalogSources = 0  # Number of astrometric catalog sources determined based upon coordinate overlap with image WCS
     foundSources = 0   # Number of sources detected in images
     matchSources = 0   # Number of sources cross matched between astrometric catalog and detected in image
```
Comment (dulude) [inline drizzlepac/hlautils/analyze.py]: Variable should not be camelcase. Rename 'matchSources' to 'match_sources' and find/replace the remaining instances.

**Comment by stsci-hack** [comment]
All comments have been addressed, along with additional related issues found in the process.  This PR will then be merged once the regression tests have run successfully.

File: drizzlepac/alignimages.py (hunk 14 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 4 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 5 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 6 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 7 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 8 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 9 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 10 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 11 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 12 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 13 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py (hunk 14 above)
Comment (stsci-hack) [reply drizzlepac/alignimages.py]: Resolved in 51e3cea

File: drizzlepac/alignimages.py [hunk 20]
```diff
@@ -1174,42 +1141,37 @@ def update_image_wcs_info(tweakwcs_output):
                 sci_ext_dict["{}".format(sciExtCtr)] = fileutil.findExtname(hdulist, 'sci', extver=sciExtCtr)
 
         # update header with new WCS info
-        updatehdr.update_wcs(hdulist,
-                             sci_ext_dict["{}".format(item.meta['chip'])],
-                             item.wcs,
-                             wcsname=wcs_name,
-                             reusename=True,
-                             verbose=True)
+        updatehdr.update_wcs(hdulist, sci_ext_dict["{}".format(item.meta['chip'])], item.wcs, wcs_name=wcs_name,
```
Comment (dulude) [inline drizzlepac/alignimages.py]:
```suggestion
        updatehdr.update_wcs(hdulist, sci_ext_dict["{}".format(item.meta['chip'])], item.wcs, wcsname=wcs_name,
```

**Pull request merged**